cheroot/server.py

```diff
@@ -2017,10 +2013,7 @@ def resolve_real_bind_addr(socket_):
 
     def tick(self):
         """Accept a new connection and put it on the Queue."""
-        if not self.ready:
-            return
```

Comment:
slight reorg here - a single check for `self.ready` at the beginning of this routine is not sufficient, since `stop()` can clear it & close the socket and connections even after this initial check.

this results in `self.connections.get_conn()` or `self.connections.expire()` failing since the selector has been closed.

instead, make `stop()` ensure that `serve()` has returned before closing down the socket and connections.